src/main/java/com/byznass/tiolktrack/kernel/dao/GpsProvider.java

```diff
@@ -1,11 +1,11 @@
 package com.byznass.tiolktrack.kernel.dao;
 
 import com.byznass.tiolktrack.kernel.TiolkTrackException;
-import com.byznass.tiolktrack.kernel.model.Gps;
+import com.byznass.tiolktrack.kernel.model.gps.Gps;
+import com.byznass.tiolktrack.kernel.model.gps.GpsWithLocations;
 
 public interface GpsProvider {
 
-	Gps getGpsById(String gpsId) throws NoGpsWithIdException, TiolkTrackException;
-
-	boolean exists(String gpsId) throws TiolkTrackException;
+	GpsWithLocations getGpsWithLocations(String userId, String name) throws NoSuchGpsException, TiolkTrackException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.kernel.dao.NoSuchGpsException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.byznass.tiolktrack.kernel.TiolkTrackException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.tiolktrack.byznass.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)